src/pal/src/sharedmemory/sharedmemory.cpp

```diff
@@ -245,7 +245,7 @@ int SharedMemoryHelpers::CreateOrOpenFile(LPCSTR path, bool createIfNotExist, bo
     _ASSERTE(!createIfNotExist || SharedMemoryManager::IsCreationDeletionFileLockAcquired());
 
     // Try to open the file
-    int openFlags = O_RDWR;
+    int openFlags = O_RDWR | O_CLOEXEC;
```

Comment:
Maybe the change should be in the `Open` function to affect all calls to it? There are a few more calls to `Open` that would not be affected by this change.

Reply:
Actually just one more call in `OpenDirectory`, the file descriptor from that is also used for a file lock in one case. Or'ing the flag in `Open` would be better, there shouldn't be a need to call `open` without that flag.

Reply:
OK. Can fix it. Although the OpenDirectory is closed immediately after. But I guess it would be more robust.

Reply:
> Although the OpenDirectory is closed immediately after.

I was referring to this code:
https://github.com/dotnet/coreclr/blob/a81680cc473a6620625815eb643c2f2fea771f2e/src/pal/src/sharedmemory/sharedmemory.cpp#L1129-L1136

That descriptor is not closed and is expected to be closed by the system when the process terminates.